### PR TITLE
Add option to always get screenshots from plotter shows

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
### Overview

This PR adds an option to always produce screenshots instead of showing the plotter when performing a `pl.show()`. We need this feature in some PyAnsys projects, where we need to have the screenshots of several plotters without modifying the underlaying code. This is related to #5194 , which we also needed for the project.

Let me know your thoughts about this feature 🙂 

### Details

I'm using UUID library to generate unique identifiers for the screenshot filenames. This may be substituted by some attribute from PyVista, but not sure if any would ensure that the name is unique.

### CODE REVIEW

File: pyvista/plotting/plotter.py [hunk 1]
```diff
@@ -6404,7 +6405,7 @@ def __init__(
                 notebook = scooby.in_ipykernel()
 
         self.notebook = notebook
-        if self.notebook:
+        if self.notebook or pyvista.SCREENSHOTS_ONLY:
```
Comment (tkoyama010) [inline pyvista/plotting/plotter.py:6408]: This is a great feature. Thank you. I will follow the `OFF_SCREEN` naming convention and suggest the variable name `ON_SCREENSHOT`.
```suggestion
        if self.notebook or pyvista.ON_SCREENSHOT:
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Review by tkoyama010 — DISMISSED** [review]
LGTM

**Comment by AlejandroFernandezLuces** [comment]
Changed to pytest `tmpfolder` for image cleanup. Also, added a fixture that resets the global variables to the values it had before running the tests, I think it's cleaner and can be reused in other tests.

**Review by MatthewFlamm — APPROVED** [review]
I can foresee users also wanting sequentially numbered filenames to keep track of which image is which, but a future PR to add in a configurable mode.

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**